pkg/cli/cmds/agent_service_windows.go

```diff
@@ -0,0 +1,136 @@
+// +build windows
+
+package cmds
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	"github.com/rancher/k3s/pkg/version"
+	"github.com/urfave/cli"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+const (
+	defaultServiceDescription = "Rancher Kubernetes Engine v2 (agent) see docs https://github.com/rancher/rke2#readme"
+)
+
+var serviceSubcommand = cli.Command{
+	Name:   "service",
+	Usage:  "Manage RKE2 as a Windows Service",
+	Action: serviceAction,
+	Flags: []cli.Flag{
+		cmds.ConfigFlag,
+		&cli.BoolFlag{
+			Name:  "add",
+			Usage: "add RKE2 as a Windows Service",
+		},
+		&cli.BoolFlag{
+			Name:  "delete",
+			Usage: "stop and delete RKE2 as a Windows Service",
+		},
+		&cli.StringFlag{
+			Name:  "service-name",
+			Usage: "name for the RKE2 service",
+			Value: version.Program,
+		},
+	},
+}
+
+func serviceAction(ctx *cli.Context) error {
+	add := ctx.Bool("add")
+	del := ctx.Bool("delete")
+	if !add && !del {
+		return errors.New("service subcommand requires --add or --delete")
+	}
+
+	if _, err := os.Stat(ctx.String("config")); os.IsNotExist(err) {
```

Comment:
it will be set as the default, im just confirming it actually exists. it's not required to pass, the default value is already there from urfave populating it from flag Value

Reply:
OK - get the intent, but with currently this file doesn't actually have to exist. For example, due to directory loading support, the user may choose to put files at c:\etc\rancher\rke2\config.yaml.d\01-my-config.yaml. I would probably just skip this check entirely.

Reply:
should we not allow/care about passing in a config at all? should we just call `rke2 agent` in the service and let everything else sort out? or you're saying if it's set then set it else leave blank?

Reply:
For this subcommand, I would:
* Leave it blank by default
* If the value is empty, don't pass through a `--config` flag to the service
* If the value is non-empty, pass it through to rke2 as `--config=X` without validating it.

Reply:
or since you're just reusing the flag including the default, just pass it through without `stat`ing I guess